bindings/tflite/java/org/tensorflow/lite/Interpreter.java

```diff
@@ -0,0 +1,436 @@
+    runForMultipleInputsOutputs(inputs, outputs);
+  }
+
+  /**
+   * Runs model inference with multiple inputs/outputs.
+   *
+   * <p>See {@link #run(Buffer, Buffer)} for buffer encoding format.
+   *
+   * <p>Runs model inference if the model takes multiple inputs, or returns multiple outputs.
+   *
+   * @param inputs an array of input {@link java.nio.Buffer}s. The inputs should be in the same
+   *     order as inputs of the model.
+   * @param outputs a map mapping output indices to {@link java.nio.Buffer}s. The caller must ensure
+   *     they are set to the appropriate write position and remaining capacity.
+   * @throws IllegalArgumentException if {@code inputs} or {@code outputs} is null or empty, in the
+   *     wrong format, or capacity.
+   * @throws IllegalStateException if inference fails.
+   */
+  public void runForMultipleInputsOutputs(
+      @NonNull Buffer[] inputs, @NonNull Map<Integer, Buffer> outputs) {
+    if (inputs == null || inputs.length == 0) {
+      throw new IllegalArgumentException("Input error: Inputs should not be null or empty.");
+    }
+    if (outputs == null || outputs.isEmpty()) {
+      throw new IllegalArgumentException("Input error: Outputs should not be null or empty.");
+    }
+
+    if (!tensorsAllocated) {
+      allocateTensors();
+    }
+
+    for (int i = 0; i < inputs.length; ++i) {
+      getInputTensor(i).copyFromBuffer(inputs[i]);
+    }
+
+    long inferenceStartNanos = System.nanoTime();
+    int status = nativeInvoke();
+    inferenceDurationNanoseconds = System.nanoTime() - inferenceStartNanos;
+
+    if (status != 0) {
+      throw new IllegalStateException(
+          String.format("Failed to run Interpreter. Returned status code: %d", status));
+    }
+
+    for (Map.Entry<Integer, Buffer> output : outputs.entrySet()) {
+      getOutputTensor(output.getKey()).copyToBuffer(output.getValue());
+    }
+  }
+
+  /**
+   * [Placeholder] for running model inference based on SignatureDef provided through @code
+   * methodName. This method is unsupported.
+   *
+   * @param inputs A Map of inputs from input name in the signatureDef to an input object.
+   * @param outputs a map mapping from output name in SignatureDef to output data.
+   * @param methodName The exported method name identifying the SignatureDef.
+   * @throws UnsupportedOperationException since IREE doesn't support tensor signatures.
+   */
+  public void runSignature(
+      @NonNull Map<String, Object> inputs,
+      @NonNull Map<String, Object> outputs,
+      String methodName) {
+    throw new UnsupportedOperationException("IREE does not support tensor signatures.");
+  }
+
+  /* Same as {@link Interpreter#runSignature(Object, Object, String)} but doesn't require
+   * passing a methodName, assuming the model has one SignatureDef. If the model has more than
+   * one SignatureDef it will throw an exception.
+   *
+   * * <p>WARNING: This is an experimental API and subject to change.
+   * */
```

Comment:
nit: remove extra `*` characters?

```suggestion
  /* Same as {@link Interpreter#runSignature(Object, Object, String)} but doesn't require
   * passing a methodName, assuming the model has one SignatureDef. If the model has more than
   * one SignatureDef it will throw an exception.
   *
   * <p>WARNING: This is an experimental API and subject to change.
   */
```

Reply:
Actually, I thought I'd removed this. I've deleted it since its not supported.